pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -292,16 +292,22 @@ func (a *Actuator) Exists(ctx context.Context, machine *machinev1beta1.Machine)
 		return false, a.clearMachineAddresses(ctx, machine)
 	}
 
+	// FIXME(rdo): This is a temporary workaround to handle states that were removed by
+	// metal3-io/baremetal-operator/pull/388. If a 4.6 cluster is upgraded with nodes
+	// in a state that isn't handled by the BMO, we may experience unexpected behaviour.
+	var stateCheck interface{} = host.Status.Provisioning.State
+	if stateCheckStr, ok := stateCheck.(string); ok {
+		if stateCheckStr == "registration error" || stateCheckStr == "power management error" {
+			log.Printf("Machine %v exists but Host is not manageable (deprecated state).", machine.Name)
+			return true, nil
+		}
+	}
+
 	switch host.Status.Provisioning.State {
 	case bmh.StateProvisioned, bmh.StateExternallyProvisioned, bmh.StateUnmanaged:
 		log.Printf("Machine %v exists.", machine.Name)
 		return true, nil
 	case bmh.StateRegistering:
-		// This case will no longer need to be handled once the changes proposed
-		// in https://github.com/metal3-io/baremetal-operator/pull/388 are
-		// available in the baremetal-operator.
```

Comment:
Why delete this comment? Without it we will be carrying this code forever.

Reply:
@zaneb perhaps a misunderstanding on my behalf, but as we revendored with a new update of the BMO code, these states no longer existed, and therefore seemed pointless to keep the comment given that in that code https://github.com/metal3-io/baremetal-operator/pull/388 had landed. I can follow up with another PR to reinstate the comment if you feel it necessary.

Reply:
@rdoxenham The comment applied to all 3 states, including StateRegistering. So the code below is now redundant but there's no reminder to us that it can be deleted.